iiod/ops.c

```diff
@@ -428,14 +428,22 @@ static ssize_t send_data(struct DevEntry *dev, struct ThdEntry *thd, size_t len)
 		unsigned int i;
 		char buf[129], *ptr = buf;
 		uint32_t *mask = demux ? thd->mask : dev->mask;
-		ssize_t ret;
+		ssize_t ret, len;
 
+		len = sizeof(buf);
 		/* Send the current mask */
 		for (i = dev->nb_words; i > 0 && ptr < buf + sizeof(buf);
-				i--, ptr += 8)
-			sprintf(ptr, "%08x", mask[i - 1]);
+				i--, ptr += 8) {
+			iio_snprintf(ptr, len, "%08x", mask[i - 1]);
+			len -= 8;
+		}
 
 		*ptr = '\n';
+		len--;
+
+		if (len < 0)
+			IIO_ERROR("send_data: string length error\n");
+
```

Comment:
I would return in case len < 0 but not sure which error code is appropriate.

Reply:
I added a `return -ENOSPC` (No space left on device)